app/src/main/java/ch/epfl/skysync/screens/admin/AddUser.kt

```diff
@@ -116,51 +116,28 @@ fun AddUserScreen(navController: NavController) {
                 emailError = !validateEmail(emailValue)
                 firstNameError = textInputValidation(firstNameValue)
                 lastNameError = textInputValidation(lastNameValue)
-                roleError = textInputValidation(roleValue)
+                roleError = !inputNonNullValidation(roleValue)
                 balloonQualificationError =
-                    if (roleValue === "Pilot") dropDownInputValidation(balloonQualificationValue)
+                    if (roleValue === UserRole.PILOT)
+                        dropDownInputValidation(balloonQualificationValue)
                     else false
                 isError =
-                    hasNoError(
+                    hasError(
                         roleError,
                         firstNameError,
                         lastNameError,
                         emailError,
                         balloonQualificationError)
                 if (!isError) {
+                  val tmpUser =
+                      TempUser(
+                          email = emailValue,
+                          firstname = firstNameValue,
+                          lastname = lastNameValue,
+                          userRole = roleValue!!,
+                          balloonQualification = balloonQualificationValue)
 
-                  /*
-                  val user: User
-                  when (roleValue) {
-                    "Admin" -> {
-                      user =
-                          Admin(
-                              firstname = firstNameValue,
-                              lastname = lastNameValue,
-                              assignedFlights = FlightGroupCalendar(),
-                              availabilities = AvailabilityCalendar())
-                    }
-                    "Pilot" -> {
-                      user =
-                          Pilot(
-                              firstname = firstNameValue,
-                              lastname = lastNameValue,
-                              assignedFlights = FlightGroupCalendar(),
-                              availabilities = AvailabilityCalendar(),
-                              qualification = balloonQualificationValue!!)
-                    }
-                    "Crew" -> {
-                      user =
-                          Crew(
-                              firstname = firstNameValue,
-                              lastname = lastNameValue,
-                              assignedFlights = FlightGroupCalendar(),
-                              availabilities = AvailabilityCalendar())
-                    }
-                  }
-
-                     */
-                  // TODO: Add user to the database
+                  viewModel.addUser(tmpUser)
```

Comment:
If I'm not mistaken there is no snackbar of feedback text that confirm that the user has been created ? It could be nice to have

Reply:
I have added it.